web/docs/auth/social-auth/keycloak.md

```diff
@@ -0,0 +1,547 @@
+import UsingAuthNote from './\_using-auth-note.md';
+import WaspFileStructureNote from './\_wasp-file-structure-note.md';
+import GetUserFieldsType from './\_getuserfields-type.md';
+import ApiReferenceIntro from './\_api-reference-intro.md';
+import UserSignupFieldsExplainer from '../\_user-signup-fields-explainer.md';
+
+Wasp supports Keycloak Authentication out of the box.
+
+[Keycloak](https://www.keycloak.org/) is an open-source identity and access management solution for modern applications and services. Keycloak provides both SAML and OpenID protocol solutions. It also has a very flexible and powerful administration UI.
+
+Let's walk through enabling Keycloak authentication, explain some of the default settings, and show how to override them.
+
+## Setting up Keycloak Auth
+
+Enabling Keycloak Authentication comes down to a series of steps:
+
+1. Enabling Keycloak authentication in the Wasp file.
+1. Adding the `User` entity.
+1. Creating a Keycloak client.
+1. Adding the necessary Routes and Pages
+1. Using Auth UI components in our Pages.
+
+<WaspFileStructureNote />
+
+### 1. Adding Keycloak Auth to Your Wasp File
+
+Let's start by properly configuring the Auth object:
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```wasp title="main.wasp"
+app myApp {
+  wasp: {
+    version: "^0.13.0"
+  },
+  title: "My App",
+  auth: {
+    // 1. Specify the User entity (we'll define it next)
+    // highlight-next-line
+    userEntity: User,
+    methods: {
+      // 2. Enable Keycloak Auth
+      // highlight-next-line
+      keycloak: {}
+    },
+    onAuthFailedRedirectTo: "/login"
+  },
+}
+```
+
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```wasp title="main.wasp"
+app myApp {
+  wasp: {
+    version: "^0.13.0"
+  },
+  title: "My App",
+  auth: {
+    // 1. Specify the User entity (we'll define it next)
+    // highlight-next-line
+    userEntity: User,
+    methods: {
+      // 2. Enable Keycloak Auth
+      // highlight-next-line
+      keycloak: {}
+    },
+    onAuthFailedRedirectTo: "/login"
+  },
+}
+```
+
+</TabItem>
+</Tabs>
+
+`userEntity` is explained in [the social auth overview](../../auth/social-auth/overview#social-login-entity).
+
+### 2. Adding the User Entity
+
+Let's now define the `app.auth.userEntity` entity:
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```wasp title="main.wasp"
+// ...
+// 3. Define the User entity
+// highlight-next-line
+entity User {=psl
+    id          Int     @id @default(autoincrement())
+    // ...
+psl=}
+```
+
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```wasp title="main.wasp"
+// ...
+// 3. Define the User entity
+// highlight-next-line
+entity User {=psl
+    id          Int     @id @default(autoincrement())
+    // ...
+psl=}
+```
+
+</TabItem>
+</Tabs>
+
+### 3. Creating a Keycloak Client
+
+1. Login into your Keycloak admin console.
+1. Under "Clients", click on "Create Client".
```

Comment:
In English, the quotation marks go inside the quotation marks:

```suggestion
1. Under "Clients," click on "Create Client."
```

It doesn't make any sense, I know, but that's the rule.

Reply:
I've decided to bold it instead of putting it in quotes. It avoids this situation and that's what we did in the other guides.